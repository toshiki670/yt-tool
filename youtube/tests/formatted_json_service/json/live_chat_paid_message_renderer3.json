{
  "replayChatItemAction": {
    "actions": [
      {
        "clickTrackingParams": "CAEQl98BIhMI7s3kp-fEigMVLvNMAh0VThWL",
        "addChatItemAction": {
          "item": {
            "liveChatPaidMessageRenderer": {
              "id": "ChwKGkNJWHF5YUhueElvREZlN1B3Z1FkN2tNQV9n",
              "timestampUsec": "1735194518874148",
              "authorName": { "simpleText": "投稿者名" },
              "authorPhoto": {
                "thumbnails": [
                  {
                    "url": "https://yt4.ggpht.com/WxTopCWfI9ikDj5r9cW9LCjyYVxsFLWYo7pO1YabJNy6cJ6vhRr5HfC5Y1fagVj45OpSfE5s8Q=s32-c-k-c0x00ffffff-no-rj",
                    "width": 32,
                    "height": 32
                  },
                  {
                    "url": "https://yt4.ggpht.com/WxTopCWfI9ikDj5r9cW9LCjyYVxsFLWYo7pO1YabJNy6cJ6vhRr5HfC5Y1fagVj45OpSfE5s8Q=s64-c-k-c0x00ffffff-no-rj",
                    "width": 64,
                    "height": 64
                  }
                ]
              },
              "purchaseAmountText": { "simpleText": "¥1,000" },
              "message": {
                "runs": [
                  {
                    "text": "メッセージです"
                  }
                ]
              },
              "headerBackgroundColor": 4294947584,
              "headerTextColor": 3741319168,
              "bodyBackgroundColor": 4294953512,
              "bodyTextColor": 3741319168,
              "authorExternalChannelId": "UCdEhG3ljd_GhzzRay0i-0nQ",
              "authorNameTextColor": 2315255808,
              "contextMenuEndpoint": {
                "clickTrackingParams": "CAgQ7rsEIhMI7s3kp-fEigMVLvNMAh0VThWL",
                "commandMetadata": {
                  "webCommandMetadata": { "ignoreNavigation": true }
                },
                "liveChatItemContextMenuEndpoint": {
                  "params": "Q2g0S0hBb2FRMGxZY1hsaFNHNTRTVzlFUm1VM1VIZG5VV1EzYTAxQlgyY2FLU29uQ2hoVlF6WnZURGhuWlV0TlpYbG5iUzAwT1dsTWNuRmxabmNTQzNCQ2FrbFpSRGwyTjFGaklBRW9CRElhQ2hoVlEyUkZhRWN6Ykdwa1gwZG9lbnBTWVhrd2FTMHdibEU0QWtnQVVBOCUzRA=="
                }
              },
              "timestampColor": 2147483648,
              "contextMenuAccessibility": {
                "accessibilityData": { "label": "Chat actions" }
              },
              "trackingParams": "CAgQ7rsEIhMI7s3kp-fEigMVLvNMAh0VThWL",
              "authorBadges": [
                {
                  "liveChatAuthorBadgeRenderer": {
                    "icon": { "iconType": "MODERATOR" },
                    "tooltip": "Moderator",
                    "accessibility": {
                      "accessibilityData": { "label": "Moderator" }
                    }
                  }
                },
                {
                  "liveChatAuthorBadgeRenderer": {
                    "customThumbnail": {
                      "thumbnails": [
                        {
                          "url": "https://yt3.ggpht.com/pXR9awenP6d6R834AgGxte9GJkrUIH_JEhTQhshA55tMmthEV8smNV8GFUhqgnNAvQEaNSml5EQhIzM=s16-c-k",
                          "width": 16,
                          "height": 16
                        },
                        {
                          "url": "https://yt3.ggpht.com/pXR9awenP6d6R834AgGxte9GJkrUIH_JEhTQhshA55tMmthEV8smNV8GFUhqgnNAvQEaNSml5EQhIzM=s32-c-k",
                          "width": 32,
                          "height": 32
                        }
                      ]
                    },
                    "tooltip": "Member (2 months)",
                    "accessibility": {
                      "accessibilityData": { "label": "Member (2 months)" }
                    }
                  }
                }
              ],
              "textInputBackgroundColor": 822083583,
              "creatorHeartButton": {
                "creatorHeartViewModel": {
                  "creatorThumbnail": {
                    "sources": [
                      {
                        "url": "https://yt3.ggpht.com/eiArqR7kKFImTZjkBW3w2SoWpMhPJXxWhmjF219v10QLXy3DLbGk8x1ag7I_eF6HGrcU3RspDA=s48-c-k-c0x00ffffff-no-rj"
                      }
                    ]
                  },
                  "heartedIcon": {
                    "sources": [
                      { "clientResource": { "imageName": "full_heart-filled" } }
                    ]
                  },
                  "unheartedIcon": {
                    "sources": [
                      { "clientResource": { "imageName": "full_heart" } }
                    ],
                    "processor": {
                      "borderImageProcessor": {
                        "imageTint": { "color": 3741319168 }
                      }
                    }
                  },
                  "heartedHoverText": "❤ by チャンネル名",
                  "heartedAccessibilityLabel": "Remove heart",
                  "unheartedAccessibilityLabel": "Heart",
                  "engagementStateKey": "EktsaXZlLWNoYXQtbWVzc2FnZS1lbmdhZ2VtZW50LXN0YXRlLUNod0tHa05KV0hGNVlVaHVlRWx2UkVabE4xQjNaMUZrTjJ0TlFWOW4gLCgB"
                }
              },
              "isV2Style": true,
              "replyButton": {
                "pdgReplyButtonViewModel": {
                  "replyButton": {
                    "buttonViewModel": {
                      "iconName": "CHAT",
                      "onTap": {
                        "innertubeCommand": {
                          "clickTrackingParams": "CAkQ68ENIhMI7s3kp-fEigMVLvNMAh0VThWL",
                          "showEngagementPanelEndpoint": {
                            "identifier": {
                              "surface": "ENGAGEMENT_PANEL_SURFACE_LIVE_CHAT",
                              "tag": "PAreply_thread"
                            },
                            "globalConfiguration": {
                              "params": "ggmcAgojVWd5dmFXOXVFanJUNjhGeXFQNTRBYUFCRHFnQjZJamNuUUkSVgopKicKGFVDNm9MOGdlS01leWdtLTQ5aUxycWVmdxILcEJqSVlEOXY3UWMQARgEIAE6I1VneXZhVzl1RWpyVDY4RnlxUDU0QWFBQkRxZ0I2SWpjblFJGm4IABgAIABKCggBEAAYACAAMABQ9b6XqtnEigNYA3gAogEAqgEMEAAaACIAKgQIABAAsAEAwAEAyAH1vpeq2cSKA-IBDAju0bO7BhD0quT_AugBAPABAPgBAIgCAJACAJoCDAiX77O7BhClufSiAyIpKicKGFVDNm9MOGdlS01leWdtLTQ5aUxycWVmdxILcEJqSVlEOXY3UWMoATAA"
                            },
                            "engagementPanelPresentationConfigs": {
                              "engagementPanelPopupPresentationConfig": {
                                "popupType": "PANEL_POPUP_TYPE_DIALOG"
                              }
                            }
                          }
                        }
                      },
                      "accessibilityText": "Reply",
                      "style": "BUTTON_VIEW_MODEL_STYLE_CUSTOM",
                      "trackingParams": "CAkQ68ENIhMI7s3kp-fEigMVLvNMAh0VThWL",
                      "type": "BUTTON_VIEW_MODEL_TYPE_TONAL",
                      "buttonSize": "BUTTON_VIEW_MODEL_SIZE_XSMALL",
                      "customBackgroundColor": 218103808,
                      "customFontColor": 3741319168,
                      "onVisible": {
                        "innertubeCommand": {
                          "clickTrackingParams": "CAkQ68ENIhMI7s3kp-fEigMVLvNMAh0VThWL",
                          "logFlowLoggingEventCommand": {
                            "flowEventType": 1,
                            "flowEventNamespace": "FLOW_EVENT_NAMESPACE_LIVE_CHAT_ENGAGEMENT",
                            "flowType": "FLOW_TYPE_LIVE_CHAT_ENGAGEMENT",
                            "flowEventMetadata": {
                              "liveChatEngagementContext": {
                                "entryPointType": "LIVE_CHAT_ENGAGEMENT_ENTRY_POINT_REPLY_BUTTON"
                              }
                            }
                          }
                        }
                      },
                      "loggingDirectives": {
                        "trackingParams": "CAkQ68ENIhMI7s3kp-fEigMVLvNMAh0VThWL",
                        "visibility": { "types": "12" }
                      }
                    }
                  },
                  "replyCountEntityKey": "EiNVZ3l2YVc5dUVqclQ2OEZ5cVA1NEFhQUJEcWdCNklqY25RSSDxAygB",
                  "replyCountPlaceholder": {
                    "content": "Reply",
                    "styleRuns": [{ "startIndex": 0, "length": 5 }]
                  }
                }
              }
            }
          },
          "clientId": "CIXqyaHnxIoDFe7PwgQd7kMA_g"
        }
      }
    ]
  },
  "videoOffsetTimeMsec": "3752720",
  "isLive": true
}
