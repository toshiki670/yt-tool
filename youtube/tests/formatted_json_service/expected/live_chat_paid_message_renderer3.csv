id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNJWHF5YUhueElvREZlN1B3Z1FkN2tNQV9n,UCdEhG3ljd_GhzzRay0i-0nQ,2024-12-26T15:28:38.874148+09:00,Paid message,投稿者名,"message: 'メッセージです'; purchaseAmount: '¥1,000'; Moderator; membershipMonths: 'Member (2 months)'"
